### DISCUSSION TITLE
Solving common installation issues

### DISCUSSION THREAD

**Discussion by nikolasibalic** [nikolasibalic/ARC-Alkali-Rydberg-Calculator#101] [General]
Place for all common installation issues encountered by users. Hopefully collective intelligence can resolve them all.

**Comment by nikolasibalic** [comment]
If `numpy` reports error after installation when you try to use ARC, and you are using conda installation/Anaconda Python distribution, the cause is most probably that curent conda installation is using older version of the Numpy, while the ARC has been compiled with the latest version of the Numpy. Problem is resolved by installing new Numpy version directly from pip. See issues #25, #63, and #100 .

**UPDATE**: As of ver. 3.1.9. This should be resolved with no extra action needed by the user (see discussion in #100 if you are interested in details of this resolution).

**Reply by Neo-Niu-XG** [reply]
It reports error when I installing ARC. I insure that my numpy is the latest version. I'm using Anaconda in macOS. Can you help me?
The error is just like this:
`Building wheel for ARC-Alkali-Rydberg-Calculator (setup.py) ... error
  ERROR: Command errored out with exit status 1:
   command: /Users/apple/opt/anaconda3/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/private/var/folders/1t/vjkw9s6n56b84fq1gy5n287r0000gn/T/pip-install-p_fp92wc/arc-alkali-rydberg-calculator/setup.py'"'"'; __file__='"'"'/private/var/folders/1t/vjkw9s6n56b84fq1gy5n287r0000gn/T/pip-install-p_fp92wc/arc-alkali-rydberg-calculator/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /private/var/folders/1t/vjkw9s6n56b84fq1gy5n287r0000gn/T/pip-wheel-beeivddq
       cwd: /private/var/folders/1t/vjkw9s6n56b84fq1gy5n287r0000gn/T/pip-install-p_fp92wc/arc-alkali-rydberg-calculator/
  Complete output (107 lines):
  running bdist_wheel
  running build
  running build_py
  creating build
  creating build/lib.macosx-10.9-x86_64-3.8
  creating build/lib.macosx-10.9-x86_64-3.8/arc
  copying arc/alkali_atom_functions.py -> build/lib.macosx-10.9-x86_64-3.8/arc
  copying arc/wigner.py -> build/lib.macosx-10.9-x86_64-3.8/arc
  copying arc/web_functionality.py -> build/lib.macosx-10.9-x86_64-3.8/arc
  copying arc/calculations_atom_single.py -> build/lib.macosx-10.9-x86_64-3.8/arc
  copying arc/beta.py -> build/lib.macosx-10.9-x86_64-3.8/arc
  copying arc/divalent_atom_data.py -> build/lib.macosx-10.9-x86_64-3.8/arc
  copying arc/materials.py -> build/lib.macosx-10.9-x86_64-3.8/arc
  copying arc/calculations_atom_pairstate.py -> build/lib.macosx-10.9-x86_64-3.8/arc
  copying arc/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/arc
  copying arc/setupc.py -> build/lib.macosx-10.9-x86_64-3.8/arc
  copying arc/arc_c_extensions.py -> build/lib.macosx-10.9-x86_64-3.8/arc
  copying arc/alkali_atom_data.py -> build/lib.macosx-10.9-x86_64-3.8/arc
  copying arc/divalent_atom_functions.py -> build/lib.macosx-10.9-x86_64-3.8/arc
  creating build/lib.macosx-10.9-x86_64-3.8/arc/advanced
  copying arc/advanced/__init__.py -> build/lib.macosx-10.9-x86_64-3.8/arc/advanced
  copying arc/advanced/population_lifetime.py -> build/lib.macosx-10.9-x86_64-3.8/arc/advanced
  creating build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/h_NIST_level_data.ascii -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/NIST_cs_quadrupole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/sodium_literature_dme.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/h_dipole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/na23_hfs_data.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/li7_dipole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/potassium_literature_dme.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/li7_quadrupole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/ca_quadrupole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/NIST_li6_quadrupole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/na_NIST_level_data.ascii -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/yb_quadrupole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/cs_dipole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/precalculated_stark.db -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/li6_quadrupole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/precalculated6j.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/rb85_hfs_data.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/k41_hfs_data.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/k_quadrupole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/caesium_literature_dme.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/li_NIST_level_data.ascii -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/angularMatrix_meta.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/li6_dipole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/NIST_k_quadrupole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/precalculated_pair.db -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/rb_NIST_level_data.ascii -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/k40_hfs_data.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/rb_quadrupole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/na23_dipole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/NIST_rb_dipole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/NIST_li7_quadrupole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/sr_dipole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/h_quadrupole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/na23_quadrupole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/yb_dipole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/NIST_na23_dipole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/NIST_li6_dipole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/li7_hfs_data.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/cs_NIST_level_data.ascii -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/NIST_na23_quadrupole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/ca_dipole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/k_dipole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/ytterbium_literature_dme.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/ca_level_data.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/strontium_literature_dme.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/cs_quadrupole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/li6_hfs_data.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/sr_quadrupole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/rb87_hfs_data.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/sr_level_data.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/k_NIST_level_data.ascii -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/NIST_cs_dipole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/NIST_rb_quadrupole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/NIST_k_dipole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/cs_hfs_data.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/angularMatrix.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/rubidium_literature_dme.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/calcium_literature_dme.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/NIST_li7_dipole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/rb_dipole_matrix_elements.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/precalculated3j.npy -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/k39_hfs_data.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  copying arc/data/yb_level_data.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data
  creating build/lib.macosx-10.9-x86_64-3.8/arc/data/refractive_index_data
  copying arc/data/refractive_index_data/Querry-o.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data/refractive_index_data
  copying arc/data/refractive_index_data/Mathar-2.8.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data/refractive_index_data
  copying arc/data/refractive_index_data/Mathar-4.35.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data/refractive_index_data
  copying arc/data/refractive_index_data/Mathar-1.3.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data/refractive_index_data
  copying arc/data/refractive_index_data/Mathar-7.5.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data/refractive_index_data
  copying arc/data/refractive_index_data/Querry-e.csv -> build/lib.macosx-10.9-x86_64-3.8/arc/data/refractive_index_data
  copying arc/arc_c_extensions.c -> build/lib.macosx-10.9-x86_64-3.8/arc
  running build_ext
  building 'arc.arc_c_extensions' extension
  Warning: Can't read registry to find the necessary compiler setting
  Make sure that Python modules winreg, win32api or win32con are installed.
  INFO: C compiler: gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/apple/opt/anaconda3/include -arch x86_64 -I/Users/apple/opt/anaconda3/include -arch x86_64
  
  creating build/temp.macosx-10.9-x86_64-3.8
  creating build/temp.macosx-10.9-x86_64-3.8/arc
  INFO: compile options: '-I/Users/apple/opt/anaconda3/lib/python3.8/site-packages/numpy/core/include -I/Users/apple/opt/anaconda3/include/python3.8 -c'
  extra options: '-Wall -O3'
  INFO: gcc: arc/arc_c_extensions.c
  xcrun: error: invalid active developer path (/Library/Developer/CommandLineTools), missing xcrun at: /Library/Developer/CommandLineTools/usr/bin/xcrun
  error: Command "gcc -Wno-unused-result -Wsign-compare -Wunreachable-code -DNDEBUG -g -fwrapv -O3 -Wall -Wstrict-prototypes -I/Users/apple/opt/anaconda3/include -arch x86_64 -I/Users/apple/opt/anaconda3/include -arch x86_64 -I/Users/apple/opt/anaconda3/lib/python3.8/site-packages/numpy/core/include -I/Users/apple/opt/anaconda3/include/python3.8 -c arc/arc_c_extensions.c -o build/temp.macosx-10.9-x86_64-3.8/arc/arc_c_extensions.o -Wall -O3" failed with exit status 1
  ERROR: Failed building wheel for ARC-Alkali-Rydberg-Calculator`

**Reply by nikolaqm** [reply]
@Neo-Niu-XG

Hi, the problem is not with numpy. Problem is that pip doesn't pick wheel (precompiled version) for MacOS python 3.8, maybe because wheel is build with newer OS version than what you have.
However since you are on Mac this should be relatively easily solvable by installing appropriate developer tools on Mac (so that you can build wheel on your computer, as the pip is trying to do, but currently fails). Can you try running from command line:

`xcode-select --install`

This should install command line tools. And try running `pip install...` after (you might need to close terminal window and open a fresh one before running pip install). That should solve the immediate error that you have now.

Please let me know if installation works then or you get a new error now.

**Reply by Neo-Niu-XG** [reply]
The new error is:
`NameError                                 Traceback (most recent call last)
<ipython-input-5-52dd19dcfcca> in <module>
----> 1 xcode-select --install

NameError: name 'xcode' is not defined
`
I'm using anaconda jupyter

**Reply by nikolaqm** [reply]
@Neo-Niu-XG   Just to clarify, you should just use terminal to install Xcode Command Line Tools.
That tools don't have to do anything with Jupyter or Anaconda, it just installs necessary commands on Mac Command line that enables compilation etc.

1. Go to Terminal in /Applications/Utilities/.
2. Input the following command string in Terminal: `xcode-select —install`
3. Follow prompts to finish installation

Let me know if it works

**Reply by Neo-Niu-XG** [reply]
It works! That's very kind of you to help me patiently! Thank you! Wish you happy!
